### PR TITLE
DX: Upgrade PHP CS Fixer and it's config

### PR DESCRIPTION
**Pull request by keradus**
ref #1555

### CODE REVIEW

File: .php_cs.dist [hunk 1]
```diff
@@ -10,12 +10,10 @@ $config = PhpCsFixer\Config::create()
     ->setRules([
         '@PSR2' => true,
         '@Symfony' => true,
-        'psr0' => false,
```
Comment (keradus) [inline .php_cs.dist]: rule was not enabled, no need to extra disable it

File: .php_cs.dist [hunk 2]
```diff
@@ -10,12 +10,10 @@ $config = PhpCsFixer\Config::create()
...
         'single_blank_line_before_namespace' => false,
         'ordered_imports' => true,
         'array_syntax' => ['syntax' => 'short'],
         'phpdoc_order' => true,
-        'blank_line_after_namespace' => true,
```
Comment (keradus) [inline .php_cs.dist]: rule was enabled by enabling `@PSR2`, no need for enabling it again

File: .php_cs.dist (hunk 1 above)
Comment (ruflin) [reply .php_cs.dist]: I like to have also rules inside that we disable to be aware of it and in case a default changes we it does not change here.

Comment (keradus) [reply .php_cs.dist]: do you really expect anyone to enable PSR-0 by default nowadays ?

File: env/elastica/Lint [hunk 3]
```diff
@@ -33,7 +33,7 @@ ENV PATH=/root/composer/vendor/bin:$PATH
 COPY composer.json /root/composer/
 
 RUN composer global remove --no-update phpdocumentor/phpdocumentor -vvv
-RUN composer global require --no-update friendsofphp/php-cs-fixer:^2.0
+RUN composer global require --no-update friendsofphp/php-cs-fixer:^2.14
```
[the remark this answers is not in the document]
Comment (keradus) [reply env/elastica/Lint]: <2.12 is not maintained for looong time.
2.14 is first one supporting PHP 7.3

[the remark this answers is not in the document]
Comment (p365labs) [reply env/elastica/Lint]: @keradus clear that's sound reasonable :) 
and thank u for supporting PHP 7.3 we made some tricks in order to let php_cs_fixer working properly on Travis.

File: .php_cs.dist (hunk 1 above)
Comment (ruflin) [reply .php_cs.dist]: Just had to look up the current state of PSR-0 and as it's deprecated, agree that we can remove it.

**Comment by p365labs** [comment]
@ruflin 👍 to merge it as with php_cs_fixer we get support for PHP 7.3

**Comment by keradus** [comment]
@ruflin , `.dist` approach is the same as for `phpunit.xml` vs `phpunit.xml.dist`

`.dist` is distributed config that is shared via repository
non-`.dist` is local config (ignored by vsc) adjusted by user, if he wanted, for example he can change cache or sth just for his local machine, without pushing it to all users

**Comment by ruflin** [comment]
@keradus Thanks for the details, appreciate it.

**Pull request merged**